listener/kcp/reconciler_factory.go

```diff
@@ -130,7 +134,8 @@ func newKcpReconciler(opts ReconcilerOpts, restcfg *rest.Config, newDiscoveryFac
 		return nil, errors.Join(ErrCreateDiscoveryClient, err)
 	}
 
+	// TODO: pass logger here as well if needed by APIBindingReconciler?
```

Comment:
yes, we need to pass the log, because that APIBindingReconciler has Reconcile func and r.log is being called there. So we will hit nil pointer exception.